src/app/upgrade.cpp

```diff
@@ -82,3 +82,36 @@ bool upgrade(const bool /*ask*/)
     exportWebUIHttpsFiles();
     return true;
 }
+
+void handleChangedDefaults(const DefaultPreferencesMode mode)
+{
+    struct DefaultValue
+    {
+        QVariant legacy;
+        QVariant current;
+
+        QVariant operator[](const DefaultPreferencesMode mode) const
+        {
+            switch (mode)
+            {
+            case DefaultPreferencesMode::Legacy:
+                return legacy;
+            case DefaultPreferencesMode::Current:
+                return current;
+            default:
+                Q_ASSERT(false);
+                return current;
+            }
+        }
+    };
+
+    const QHash<QString, DefaultValue> changedDefaults {
```

Comment:
Since we are not really using the advantages of a hash table, I would use a `QVector` here and add an additional field to `struct DefaultValue` which store the setting path (e.g. "BitTorrent/Session/QueueingSystemEnabled").
As I already give my approval, this suggestion is purely optional.

Reply:
Changed.